### PR TITLE
Add support for using Aryn DocParse chunking from `aryn-sdk`

### PR DESCRIPTION
**Pull request by MarkLindblad**

### CODE REVIEW

File: lib/aryn-sdk/aryn_sdk/partition/partition.py [hunk 1]
```diff
@@ -101,6 +118,8 @@ def partition_file(
     if aryn_config is None:
         aryn_config = ArynConfig()
 
+    docparse_url = aps_url or docparse_url  # For backwards compatibility
```
Comment (HenryL27) [inline lib/aryn-sdk/aryn_sdk/partition/partition.py:121]: would prob change docparse_url to `Optional[str]=None` and write this as 
```python
if aps_url is not None:
    if docparse_url is not None:
        logging.warning("\"aps_url\" and \"docparse_url\" parameters were both set. \"aps_url\" is deprecated. Using \"docparse_url\".")
    else:
        logging.warning("\"aps_url\" parameter is deprecated. Use \"docparse_url\" instead")
        docparse_url = aps_url
if docparse_url is None:
    docparse_url = ARYN_DOCPARSE_URL
assert docparse_url is not None, "wtf"
```

File: lib/aryn-sdk/aryn_sdk/partition/partition.py [hunk 2]
```diff
@@ -183,6 +203,7 @@ def _json_options(
     extract_images: bool = False,
     selected_pages: Optional[list[Union[list[int], int]]] = None,
     output_format: Optional[str] = None,
+    chunking_options: Optional[dict[str, Any]] = None,
```
Comment (HenryL27) [inline lib/aryn-sdk/aryn_sdk/partition/partition.py:206]: we handle malformed chunking options well in the service, yes?

**Review by HenryL27 — APPROVED** [review]
k. Not sure about all the bwc stuff for endpoint logic - we're still in v0.x so I'd prefer to just break it unless we have something critical that depends on the param existing.
Along that vein a middle path is to "deprecate" it - I wrote some code to make that more obvious (although I'll bet there's a more idiomatic way)

File: lib/aryn-sdk/aryn_sdk/partition/partition.py (hunk 2 above)
Comment (MarkLindblad) [reply lib/aryn-sdk/aryn_sdk/partition/partition.py:206]: Yes.

File: lib/aryn-sdk/aryn_sdk/partition/partition.py (hunk 1 above)
Comment (MarkLindblad) [reply lib/aryn-sdk/aryn_sdk/partition/partition.py]: I've now implemented a variant of your suggested change.

**Pull request merged**